backend/code/archipelag/market/views.py

```diff
@@ -63,10 +63,25 @@ def get_new_market(current_ngo, body_data):
 
 
 def get_messages(request, market_id):
+
     market = Market.objects.filter(id=market_id)
+    messages = Message.objects.filter(market=market_id).all()
+    logs = []
+
+
     template_name = 'market/message_list.html'
     return render(
             request, template_name,
             {
                 'messages': Message.objects.filter(market=market).all(),
+                'logs': get_logs(messages)
             })
+
+
+def get_logs(messages):
+    for message in messages:
+        events = Event.objects.filter(id_connected_object=message.id).all()
+        if events:
+            for event in events:
+                date_created = event.date_created.strftime("%Y-%m-%d %H:%M")
+                yield "Użytkownik {} udostępnił na {} dnia {}".format(event.owner, message.type, date_created)
```

Comment:
Only one type of event logs?

Reply:
And method is get_logs but you yield events. It is ok?

Reply:
Why dont you use method `Event.get_share_log` ?

Reply:
1.in this PR I added only one type of log. 2.i changed method to ' get_logs_iterator' , 3.I forget about this method